extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/music/GoogleMusicHttpApi.java

```diff
@@ -213,22 +220,26 @@ private <T> T makePutRequest(
     // Wait for write permit before making request
     writeRateLimiter.acquire();
 
-    HttpRequestFactory requestFactory = httpTransport.createRequestFactory();
-    HttpRequest putRequest =
-        requestFactory.buildPutRequest(
-            new GenericUrl(baseUrl + "?" + generateParamsString(parameters)), httpContent);
-    putRequest.setReadTimeout(2 * 60000); // 2 minutes read timeout
+    HttpRequestFactory requestFactory = httpTransport.createRequestFactory(credential);
+    HttpRequest patchRequest =
+        requestFactory.buildPatchRequest(
+                new GenericUrl(baseUrl + "?" + generateParamsString(parameters)), httpContent)
+            // TODO: github.com/googleapis/google-http-java-client/issues/1316
+            .setRequestMethod("POST")
+            .setHeaders(new HttpHeaders().set("X-HTTP-Method-Override", "PATCH"));
+    patchRequest.setReadTimeout(2 * 60000); // 2 minutes read timeout
     HttpResponse response;
 
     try {
-      response = putRequest.execute();
+      response = patchRequest.execute();
     } catch (HttpResponseException e) {
       response =
           handleHttpResponseException(
               () ->
-                  requestFactory.buildPutRequest(
-                      new GenericUrl(baseUrl + "?" + generateParamsString(parameters)),
-                      httpContent),
+                  requestFactory.buildPatchRequest(
```

Comment:
mind factoring this out into a static method eg: `private static HttpRequest buildPatchRequest(baseUrl, parameters, httpContent)` and put the TODO on there? (and have these two blocks of code use that documented-block instead?

Reply:
Done in latest commit 7b70e28051bdc5a540f2c97e16d53578cb5601fd.